pkg/plugins/helm/parser.go

```diff
@@ -0,0 +1,200 @@
+package helm
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ghodss/yaml"
+	"k8s.io/helm/pkg/chartutil"
+	"k8s.io/helm/pkg/proto/hapi/chart"
+	_ "k8s.io/kubernetes/pkg/api/install"
+	_ "k8s.io/kubernetes/pkg/apis/apps/install"
+	_ "k8s.io/kubernetes/pkg/apis/extensions/install"
+
+	appclient "openpitrix.io/openpitrix/pkg/client/app"
+	runtimeclient "openpitrix.io/openpitrix/pkg/client/runtime"
+	"openpitrix.io/openpitrix/pkg/models"
+	"openpitrix.io/openpitrix/pkg/pb"
+)
+
+const (
+	defaultNamespace = "default"
+)
+
+type Parser struct {
+}
+
+func (p *Parser) ParseCluster(vals map[string]interface{}, versionId string) (*models.Cluster, error) {
+	clusterId, ok := p.getStringFromValues(vals, "ClusterId")
+	if !ok {
+		return nil, fmt.Errorf("ClusterId must be set and type is string")
+	}
+
+	name, ok := p.getStringFromValues(vals, "Name")
+	if !ok {
+		name = ""
+	}
+
+	desc, ok := p.getStringFromValues(vals, "Description")
+	if !ok {
+		desc = ""
+	}
+
+	ctx := context.Background()
+	appManagerClient, err := appclient.NewAppManagerClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	req := &pb.DescribeAppVersionsRequest{
+		VersionId: []string{versionId},
+	}
+
+	resp, err := appManagerClient.DescribeAppVersions(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(resp.AppVersionSet) == 0 {
+		return nil, fmt.Errorf("App version [%s] not found", versionId)
+	}
+
+	appVersion := resp.AppVersionSet[0]
+
+	runtimeId, ok := p.getStringFromValues(vals, "RuntimeId")
+	if !ok {
+		return nil, fmt.Errorf("RuntimeId must be set and type is string")
+	}
+
+	cluster := &models.Cluster{
+		ClusterId:   clusterId,
+		Name:        name,
+		Description: desc,
+		AppId:       appVersion.AppId.GetValue(),
+		VersionId:   versionId,
+		RuntimeId:   runtimeId,
+		CreateTime:  time.Now(),
+		StatusTime:  time.Now(),
+	}
+
+	return cluster, nil
+}
+
+func (p *Parser) Parse(c *chart.Chart, conf []byte, versionId string) (*models.ClusterWrapper, error) {
+	vals, err := p.parseValues(c, conf)
+	if err != nil {
+		return nil, err
+	}
+
+	cluster, err := p.ParseCluster(vals, versionId)
+	if err != nil {
+		return nil, err
+	}
+
+	clusterWrapper := &models.ClusterWrapper{
+		Cluster: cluster,
```

Comment:
ClusterNode,  ClusterRole, ClusterLinks may be needed.

Reply:
I have no idea about which concept in k8s can be mapping as ClusterRole?

Reply:
A deployment is a role, a pod is a node, 
cpu, memory, env(defined in values.yaml, each role has the same env) should be stored into ClusterRole.

Reply:
Ok